File: rest/incoming-phone-numbers/list-post-example-1/list-post-example-1.6.x.py
# Download the Python helper library from twilio.com/docs/python/install
import os
from twilio.rest import Client

# Your Account Sid and Auth Token from twilio.com/user/account
# To set up environmental variables, see http://twil.io/secure
account_sid = os.environ['TWILIO_ACCOUNT_SID']
auth_token = os.environ['TWILIO_AUTH_TOKEN']
client = Client(account_sid, auth_token)

number = client.incoming_phone_numbers \
               .create(friendly_name="My Company Line",
                       voice_url="http://demo.twilio.com/docs/voice.xml",
                       phone_number="+15105647903",
                       voice_method="GET")

print(number.sid)

File: rest/incoming-phone-numbers/list-post-example-1/list-post-example-1.6.x.cs
// Download the twilio-csharp library from twilio.com/docs/libraries/csharp
using System;
using Twilio;
using Twilio.Rest.Api.V2010.Account;
using Twilio.Types;
using Twilio.Http;

class Example
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Auth Token at twilio.com/console
        // To set up environmental variables, see http://twil.io/secure
        const string accountSid = Environment.GetEnvironmentVariable("TWILIO_ACCOUNT_SID");
        const string authToken = Environment.GetEnvironmentVariable("TWILIO_AUTH_TOKEN");
        TwilioClient.Init(accountSid, authToken);

        var number = IncomingPhoneNumberResource.Create(
            friendlyName: "My Company Line",
            voiceUrl: new Uri("http://demo.twilio.com/docs/voice.xml"),
            phoneNumber: new PhoneNumber("+15105647903"),
            voiceMethod: HttpMethod.Get);

        Console.WriteLine(number.Sid);
    }
}
